pkg/provider/kubernetes/alertmanager.go

```diff
@@ -0,0 +1,145 @@
+		), nil
+	}
+}
+
+// Get gets an Alertmanager object and Secret which contains the configuration of this Alertmanager.
+func (p *AlertmanagerProvider) Get(cluster *kubermaticv1.Cluster, userInfo *provider.UserInfo) (*kubermaticv1.Alertmanager, *corev1.Secret, error) {
+	impersonationClient, err := createImpersonationClientWrapperFromUserInfo(userInfo, p.createSeedImpersonatedClient)
+	if err != nil {
+		return nil, nil, err
+	}
+	ctx := context.Background()
+	alertmanager := &kubermaticv1.Alertmanager{}
+	if err := impersonationClient.Get(ctx, types.NamespacedName{
+		Name:      resources.AlertmanagerName,
+		Namespace: cluster.Status.NamespaceName,
+	}, alertmanager); err != nil {
+		return nil, nil, err
+	}
+	configSecret := &corev1.Secret{}
+	if err := impersonationClient.Get(ctx, types.NamespacedName{
+		Name:      alertmanager.Spec.ConfigSecret.Name,
+		Namespace: cluster.Status.NamespaceName,
+	}, configSecret); err != nil {
+		return nil, nil, err
+	}
+	return alertmanager, configSecret, nil
+}
+
+// Create only updates an Alertmanager object and corresponding config Secret since Alertmanager and Secret will
+// be created by alertmanager configuration controller.
+func (p *AlertmanagerProvider) Create(expectedAlertmanager *kubermaticv1.Alertmanager, expectedSecret *corev1.Secret, userInfo *provider.UserInfo) (*kubermaticv1.Alertmanager, *corev1.Secret, error) {
+	impersonationClient, err := createImpersonationClientWrapperFromUserInfo(userInfo, p.createSeedImpersonatedClient)
+	if err != nil {
+		return nil, nil, err
+	}
+	ctx := context.Background()
+	alertmanager := &kubermaticv1.Alertmanager{}
+
+	if err := impersonationClient.Get(ctx, types.NamespacedName{
+		Name:      expectedAlertmanager.Name,
+		Namespace: expectedAlertmanager.Namespace,
+	}, alertmanager); err != nil {
+		return nil, nil, utilerrors.New(http.StatusInternalServerError, fmt.Errorf("failed to get alertmanager: %w", err).Error())
+	}
+
+	if alertmanager.Spec.ConfigSecret.Name == "" {
+		return nil, nil, utilerrors.New(http.StatusInternalServerError, "failed to find alertmanager configuration")
+	}
+	secret := &corev1.Secret{}
+	if err := impersonationClient.Get(ctx, types.NamespacedName{
+		Name:      alertmanager.Spec.ConfigSecret.Name,
+		Namespace: alertmanager.Namespace,
+	}, secret); err != nil {
+		return nil, nil, utilerrors.New(http.StatusInternalServerError, fmt.Errorf("failed to get config secret: %w", err).Error())
+	}
+	secret.Data = expectedSecret.Data
+	if err := impersonationClient.Update(ctx, secret); err != nil {
+		return nil, nil, err
+	}
+	return alertmanager, secret, nil
+}
+
+// Delete resets corresponding config Secret of Alertmanager object to the default config. Note that Delete will not remove
+// Alertmanager object, it will only delete the config secret, and alertmanager controller will create default config secret.
+func (p *AlertmanagerProvider) Delete(cluster *kubermaticv1.Cluster, userInfo *provider.UserInfo) error {
```

Comment:
maybe the endpoint which uses this should be called reset instead of delete. Although it does delete something the purpose of it is to reset the config to default. WDYT?